engine/graphics/proto/graphics/graphics_ddf.proto

```diff
@@ -36,15 +42,28 @@ message VertexAttribute
         TYPE_FLOAT            = 7;
     }
 
+    enum ShaderType
+    {
+        SHADER_TYPE_NUMBER = 1;
+        SHADER_TYPE_VEC2   = 2;
+        SHADER_TYPE_VEC3   = 3;
+        SHADER_TYPE_VEC4   = 4;
+        SHADER_TYPE_MAT2   = 5;
+        SHADER_TYPE_MAT3   = 6;
+        SHADER_TYPE_MAT4   = 7;
+    }
```

Comment:
We have a special type for these here instead of the "ShaderDataType" that also contains these values. Not sure we can merge them, since the shaderdatatype has things like uint, float and so forth, but we already have that covered that with the "DataType" enum